raw_map/src/lane_specs.rs

```diff
@@ -1,583 +1,307 @@
-        //
-        // TODO Can we express this more simply by referring to a left_side and right_side here?
-        if tags.is_any("cycleway:right", vec!["lane", "track"]) {
-            if cfg.driving_side == DrivingSide::Right {
-                if tags.is("cycleway:right:oneway", "no") || tags.is("oneway:bicycle", "no") {
-                    fwd_side.push(back(LaneType::Biking));
+            // Fix direction on outer lanes
+            for (idx, lane) in result.iter_mut().enumerate() {
+                if lane.lt == LaneType::Sidewalk || lane.lt == LaneType::Shoulder {
+                    if idx == 0 {
+                        lane.dir = if cfg.driving_side == DrivingSide::Right {
+                            Direction::Back
+                        } else {
+                            Direction::Fwd
+                        };
+                    } else {
+                        // Assume last
+                        lane.dir = if cfg.driving_side == DrivingSide::Right {
+                            Direction::Fwd
+                        } else {
+                            Direction::Back
+                        };
+                    }
                 }
-                fwd_side.push(fwd(LaneType::Biking));
-            } else {
-                if tags.is("cycleway:right:oneway", "no") || tags.is("oneway:bicycle", "no") {
-                    back_side.push(fwd(LaneType::Biking));
-                }
-                back_side.push(back(LaneType::Biking));
-            }
-        }
-        if tags.is("cycleway:left", "opposite_lane") || tags.is("cycleway", "opposite_lane") {
-            if cfg.driving_side == DrivingSide::Right {
-                back_side.push(back(LaneType::Biking));
-            } else {
-                fwd_side.push(fwd(LaneType::Biking));
             }
+
+            result
         }
-        if tags.is_any("cycleway:left", vec!["lane", "opposite_track", "track"]) {
-            if cfg.driving_side == DrivingSide::Right {
-                if tags.is("cycleway:left:oneway", "no") || tags.is("oneway:bicycle", "no") {
-                    back_side.push(fwd(LaneType::Biking));
-                    back_side.push(back(LaneType::Biking));
-                } else if oneway {
-                    fwd_side.insert(0, fwd(LaneType::Biking));
-                } else {
-                    back_side.push(back(LaneType::Biking));
-                }
-            } else {
-                // TODO This should mimic the logic for right-handed driving, but I need test cases
-                // first to do this sanely
-                if tags.is("cycleway:left:oneway", "no") || tags.is("oneway:bicycle", "no") {
-                    fwd_side.push(back(LaneType::Biking));
-                }
-                fwd_side.push(fwd(LaneType::Biking));
-            }
+        Err(err) => {
+            error!("osm2lanes broke on {:?}: {}", orig_tags, err);
```

Comment:
I guess the tracing we are missing here is way_id

Reply:
It's awkwardly embedded in the list of tags as abst:way_id. I'll drag it out and print a URL to make for faster debugging.

There were straight-up panics though, I'm adding some debugging to get examples